examples/mytestlib.c

```diff
@@ -22,23 +22,107 @@
 #include <fcntl.h>
 #include <unistd.h>
 
-struct vaccel_arg {
-	uint32_t len;
-	void *buf;
+#include <vaccel.h>
+
+struct mydata
+{
+	uint32_t size;
+	int* array;
 };
 
+/* Serializer function for `struct mydata` */
+void* ser(void* buf, uint32_t* bytes){
+	struct mydata* non_ser = (struct mydata*)buf;
+
+	uint32_t size = (non_ser->size + 1) * sizeof(int);
+	int* ser_buf = malloc(size);
+
+	memcpy(ser_buf, (int*)(&non_ser->size), sizeof(int));
+
+	for(int i=0; i<(int)non_ser->size; i++)
+		memcpy(&ser_buf[i+1], &non_ser->array[i], sizeof(int));
+
+
+	*bytes = size;
+	return ser_buf;
+}
+
+/* Deserializer function for `struct mydata` */
+void* deser(void* buf, uint32_t bytes){
```

Comment:
`bytes` is unused in this case, since `deser` (deserializer of `struct mydata`) has access on the length of the array by reading the first 4 bytes of  the the serialized data ('buf' parameter). However, if the serialized data didn't contain the size, the deserializer wouldn't have access to it, so it could be provided as a parameter.

Reply:
I see, that is fine, we can simply use a compiler attribute, in order to remove the compiler warning
```
--- a/examples/mytestlib.c
+++ b/examples/mytestlib.c
@@ -48,7 +48,7 @@ void* ser(void* buf, uint32_t* bytes){
 }
 
 /* Deserializer function for `struct mydata` */
-void* deser(void* buf, uint32_t bytes){
+void* deser(void* buf, uint32_t __attribute__((unused)) bytes){
        int *ser_buf = (int*)buf;
        int size = ser_buf[0];
```